controller.go

```diff
@@ -234,3 +267,75 @@ func ensureNodeExists(instances cloudprovider.Instances, node *v1.Node) (bool, e
 
 	return true, nil
 }
+
+// If clusterAPI and clusterID are set, then we are running
+// inside Giantnetes, so try to get certificates and create
+// kubernetes client for guest cluster. If not proceed with
+// kubeconfig file.
+func newGuestClient(clusterID, clusterAPI, master, kubeconfig string) (clientset.Interface, error) {
+	if clusterAPI != "" && clusterID != "" {
+		return newGuestClientFromSecret(clusterAPI, clusterID)
+	}
+
+	if kubeconfig == "" {
+		err := fmt.Errorf("kubeconfig must be set, if %v and %v are not used", clusterAPIEnv, clusterIDEnv)
+		return nil, err
+	}
+
+	// Create kubeclient config from kubeconfig file.
+	config, err := clientcmd.BuildConfigFromFlags(master, kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create the client.
+	kubeClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return kubeClient, nil
+}
+
+func newGuestClientFromSecret(clusterID, clusterAPI string) (clientset.Interface, error) {
+	// Get host cluster client to retrieve certificates.
```

Comment:
Can this https://github.com/giantswarm/certificatetpr/blob/0d95e43c5d2fba49e55308ea807ed562d358c6be/service.go#L69 be used?

**If not:**

Can we wrap it in a block? That way it is clear that we only want a secret from this block.

Something like:

```
// Get host cluster client to retrieve certificates.
var secret <package>.Secret
{

	// ...

	 secret, err = hostClient.CoreV1().Secrets(metav1.NamespaceDefault).Get(secretName, metav1.GetOptions{})
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
}

Reply:
done